### ISSUE TITLE
searching for orphaned nodes very slow (performance bug)

### ISSUE THREAD

**Issue by ftes** [Docear/Desktop#500]
See http://www.docear.org/support/forums/docear-support-forums-group3/bug-reports-forum6/searching-orphaned-annotations-is-taking-ages-thread1412.0/#postid-5031.

Performance bug identified in `MonitoringWorker.searchingOrphanedNodes`.
![performance analysis](https://ibin.co/3EGuhe9g3SbO.png)

It loops over all annotations. For each annotation A it checks if A is still present in its file F.
This means that if F has 100 annotations, it is opened 100 times during `searchingOrphanedNodes`.

The loop should be "switched around" so that F is only opened once.

**Comment by F-A** [comment]
I also find it very annoying. Docear takes several minutes to start due to that.
Any chance the Pull Request will be soon approved?

**Comment by ftes** [comment]
The docear repo is no longer being maintained unfortunately.

**Comment by F-A** [comment]
Yes, this is the impression I had. I just hoped that some admin would still read the issues from time to time...

**Comment by Joeran** [comment]
i am really sorry, but in the foreseeable future, we wont have time to continue the development. would anyone of you be interested in maintaining the source code? then please send me a message with a few details about you.

**Comment by F-A** [comment]
I'm sorry, I have to little experience in JAVA to take over the project. It would take me too much time, which I do not have currently, to get used to the code...
